File: platforms/android/library/src/main/kotlin/com/hafidznoor/idnfinlogos/LogoMeta.kt
package com.hafidznoor.idnfinlogos

/**
 * Metadata for a single logo bundled in the library.
 *
 * The raw SVG file is shipped as an Android asset. Open it with:
 *   `context.assets.open(logo.assetPath)`
 * Or load it directly into an image view via Coil 2's SVG decoder:
 *   `AsyncImage(model = "file:///android_asset/${logo.assetPath}", imageLoader = svgLoader, ...)`
 */
public data class LogoMeta(
    val slug: String,
    val name: String,
    val category: String,
    val aliases: List<String>,
    val tags: List<String>,
    val assetPath: String,
)

File: platforms/flutter/lib/src/logo_meta.dart
/// Metadata for a single logo bundled in the idn_finlogos package.
///
/// The raw SVG file ships as a Flutter asset under
/// `packages/idn_finlogos/assets/idn-finlogos/<slug>.svg`. Render it with
/// `flutter_svg`:
///
/// ```dart
/// import 'package:flutter_svg/flutter_svg.dart';
/// SvgPicture.asset(logo.assetPath);
/// ```
class LogoMeta {
  final String slug;
  final String name;
  final String category;
  final List<String> aliases;
  final List<String> tags;

  const LogoMeta({
    required this.slug,
    required this.name,
    required this.category,
    required this.aliases,
    required this.tags,
  });

  /// Asset path resolvable by Flutter's asset bundle (and `flutter_svg`).
  String get assetPath => 'packages/idn_finlogos/assets/idn-finlogos/$slug.svg';

  @override
  bool operator ==(Object other) =>
      other is LogoMeta && other.slug == slug;

  @override
  int get hashCode => slug.hashCode;

  @override
  String toString() => 'LogoMeta(slug: $slug, category: $category)';
}
